pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -102,13 +103,11 @@ func (p *producerJobResumer) Resume(ctx context.Context, execCtx interface{}) er
 			case jobspb.StreamReplicationProgress_FINISHED_SUCCESSFULLY:
 				return p.releaseProtectedTimestamp(ctx, execCfg)
 			case jobspb.StreamReplicationProgress_FINISHED_UNSUCCESSFULLY:
-				return j.NoTxn().Update(ctx, func(txn isql.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
-					ju.UpdateStatus(jobs.StatusCancelRequested)
```

Comment:
i'm not entirely sure want to fail this job whenever the consumer does not succeed, but I think my approach is simpler and less confusing than cancellation behavior we had before.